plugins/modules/jenkins_node.py

```diff
@@ -0,0 +1,1311 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: jenkins_node
+short_description: Manage Jenkins nodes
+description:
+    - Manage Jenkins nodes with Jenkins REST API.
+requirements:
+  - "python-jenkins >= 0.4.12"
+author:
+  - Connor Newton (@phyrwork)
+extends_documentation_fragment:
+  - community.general.attributes
+attributes:
+  check_mode:
+    support: none
+  diff_mode:
+    support: none
+options:
+  name:
+    description:
+      - Name of the Jenkins node to manage.
+    required: yes
+    type: str
+  url:
+    description:
+      - URL of the Jenkins server.
+    default: http://localhost:8080
+    type: str
+  user:
+    description:
+      - User to authenticate with the Jenkins server.
+    type: str
+  password:
+    description:
+      - Password to authenticate with the Jenkins server.
+    type: str
+  token:
+    description:
+      - API token to authenticate with the Jenkins server.
+    type: str
+  state:
+    description:
+      - Specifies whether the Jenkins node should be V(present) (created), V(absent)
+        (deleted), enabled (online) or disabled (offline).
+    default: present
+    type: str
+    choices:
+      - enabled
+      - disabled
+      - present
+      - absent
+  description:
+    description:
+      - When specified, sets the Jenkins node description.
+    type: str
+  num_executors:
+    description:
+      - When specified, sets the Jenkins node executor count.
+    type: int
+  remote_fs:
+    description:
+      - When specified, sets the Jenkins node remote root directory.
+    type: path
+  labels:
+    description:
+      - When specified, sets the Jenkins node labels.
+    type: list
+    elements: str
+  mode:
+    description:
+      - When specified, sets the Jenkins node usage policy.
+      - V(normal) means use node as much as possible.
+      - V(exclusive) means only build jobs with label expressions that match tnode.
+    type: str
+    choices:
+      - normal
+      - exclusive
+  launcher:
+    description:
+      - When specified, sets the Jenkins node launch method.
+      - Launcher V(type) must be specified.
+      - V(args) are specified according to the launcher V(type) (see examples).
+    type: dict
+    suboptions:
+      type:
+        description:
+          - Specifies the launch method.
+        type: str
+        choices:
+          - ssh
+          - jnlp
+        required: yes
```

Comment:
Always use `true` and `false for YAML booleans.
```suggestion
        required: true
```
There are multiple more in this file.